jum.py

```diff
@@ -81,55 +84,69 @@ def get_random_empty_pos(self, img):
-                if self.overlaps_frame(img, *utils.centre_to_top_left(img, (cx, cy))):
-                    self.bgd.blit(img, self.get_random_empty_pos(img))
+                img = self.imgs[n - 1]
+                final_top_left = utils.centre_to_top_left(img, (cx, cy))
+                init_top_left = (0, g.h - img.get_height())
+                if self.overlaps_frame(img, final_top_left):
+                    self.obj_bgd.append(n)
                 else:
-                    grid.append(GridImage(img, cx, cy))
+                    self.grid_imgs.append(GridImage(img, init_top_left, final_top_left))
                 cx += self.dx
                 k += 1
             cy += self.dy
-        self.lookFor = random.sample(grid, self.find_n)
-        for img in grid:
-            if img not in self.lookFor:
-                utils.centre_blit(self.bgd, img.img, (img.cx, img.cy))
 
+        self.obj_bgd = [
+            GridImage(
+                self.imgs[n - 1],
+                (0, g.h - img.get_height()),
+                self.get_random_empty_pos(self.imgs[n - 1])
+            ) for n in self.obj_bgd
+        ]
+
+        self.lookFor = random.sample(self.grid_imgs, self.find_n)
+        self.play_anim(0)
+
+    def play_anim(self, alpha):
+        self.bgd.fill((128, 0, 0))
+        for img_list in (self.grid_imgs, self.obj_bgd):
+            for img in img_list:
+                img.x = img.from_x + (img.to_x - img.from_x) * self.easing(alpha)
+                img.y = img.from_y + (img.to_y - img.from_y) * self.easing(alpha)
+                if img not in self.lookFor:
+                    self.bgd.blit(img.img, (img.x, img.y))
 
     def draw(self):
         g.screen.blit(self.bgd, (g.sx(0), 0))
         cxy = (g.sx(30.5), g.sy(20))
         utils.display_number(g.count, cxy, g.font2, utils.CREAM)
         for img in self.lookFor:
             if not img.found:
-                utils.centre_blit(g.screen, img.img, (img.cx, img.cy))
-        if not self.complete and not g.setup_on:
-            img = self.lookFor[self.current_ind].img
-            cxy = (self.frame_cx, self.frame_cy)
-            utils.centre_blit(g.screen, self.frame, cxy)
-            utils.centre_blit(g.screen, img, cxy)
-            x, y = cxy
-            x -= self.frame.get_width() / 2
-            x += g.sy(.24)
-            y += self.frame.get_height() / 2
-            y -= g.sy(.76)
-            n = self.found
-            s = str(n) + ' / ' + str(self.find_n)
-            utils.text_blit1(g.screen, s, g.font1, (x, y), utils.BLACK, False)
+                g.screen.blit(img.img, (img.x, img.y))
+
+        img = self.lookFor[self.current_ind].img
+        cxy = (self.frame_cx, self.frame_cy)
+        utils.centre_blit(g.screen, self.frame, cxy)
+        utils.centre_blit(g.screen, img, cxy)
+        x, y = cxy
+        x -= self.frame.get_width() / 2
+        x += g.sy(.24)
+        y += self.frame.get_height() / 2
+        y -= g.sy(.76)
+        n = self.found

```

Comment:
This seems redundant, why not just use `self.found` as I can see it's only used in the next line?